BitwardenWatchWidgetExtension/BitwardenWatchWidgetExtension.swift

```diff
@@ -0,0 +1,80 @@
+import SwiftUI
+import WidgetKit
+
+/// Provider that provides the entries for the widget.
+struct Provider: TimelineProvider {
+    func placeholder(in context: Context) -> SimpleEntry {
+        SimpleEntry(date: Date())
+    }
+
+    func getSnapshot(in context: Context, completion: @escaping (SimpleEntry) -> Void) {
+        let entry = SimpleEntry(date: Date())
+        completion(entry)
+    }
+
+    func getTimeline(in context: Context, completion: @escaping (Timeline<Entry>) -> Void) {
+        let entry = SimpleEntry(date: Date())
+        let timeline = Timeline(entries: [entry], policy: .never)
+        completion(timeline)
+    }
+}
+
+/// Simple timeline entry to comply with the widget provider.
+struct SimpleEntry: TimelineEntry {
+    let date: Date
+}
+
+/// Entry view to be used to render the widget.
+struct BitwardenComplicationEntryView: View {
+    /// The widget family.
+    @Environment(\.widgetFamily) var family
+    /// The widget rendering mode.
+    @Environment(\.widgetRenderingMode) var renderingMode
+
+    /// The provider entry
+    var entry: Provider.Entry
+
+    var body: some View {
+        if renderingMode == .fullColor {
+            Image("ComplicationIcon")
+                .resizable()
+                .scaledToFit()
+                .clipShape(Circle())
+                .padding()
+        } else {
+            Image("AccentedComplicationIcon")
+                .resizable()
+                .scaledToFit()
+                .padding(5)
+                .widgetAccentable()
+        }
+    }
+}
+
+/// Extension entry point of the widget.
+@main
+struct BitwardenWatchWidgetExtension: Widget {
+    let kind: String = "BitwardenWatchWidgetExtension"
+
+    var body: some WidgetConfiguration {
+        StaticConfiguration(kind: kind, provider: Provider()) { entry in
+            if #available(watchOS 10.0, *) {
+                BitwardenComplicationEntryView(entry: entry)
+                    .containerBackground(.fill.tertiary, for: .widget)
+            } else {
+                BitwardenComplicationEntryView(entry: entry)
+                    .padding()
+                    .background()
+            }
+        }
+        .configurationDisplayName("Bitwarden")
+        .supportedFamilies([.accessoryCircular, .accessoryCorner, .accessoryInline])
+    }
+}
+
+@available(watchOS 10.0, *)
+#Preview(as: .accessoryCorner) {
+    BitwardenWatchWidgetExtension()
+} timeline: {
+    SimpleEntry(date: .now)
+}
```

Comment:
🤔 I suspect the answer is "no", but is there any way we can do snapshot tests of this?

Reply:
I'd need to check but my intention was to bring this up in the next mobile meeting given that we don't have tests in the watchOS target neither. So if we add them here we'd need to add them in the watch app as well.

Reply:
Definitely something to discuss, but I'm comfortable not doing them right now. I kinda feel like revamping the watch app and bringing it a bit in line with our naming/architecture conventions of the main app is something we should also look at in general, and that would be a good time to do that.

Reply:
Yes, totally agree. I've added a [spike](https://bitwarden.atlassian.net/browse/PM-10998) to check whether we can add the test structure to the watchOS app and the widget extension and also [another ticket](https://bitwarden.atlassian.net/browse/PM-10999) for the watchOS app architecture/code guidelines update.